src/Microsoft.Data.Analysis/DataFrame.Arrow.cs

```diff
@@ -3,18 +3,126 @@
+                    dataFrameColumn = new ArrowStringDataFrameColumn(field.Name, dataMemory, offsetsMemory, nullMemory, stringArray.Length, stringArray.NullCount);
+                    break;
+                case ArrowTypeId.UInt8:
+                    PrimitiveArray<byte> arrowbyteArray = (PrimitiveArray<byte>)arrowArray;
+                    ReadOnlyMemory<byte> byteValueBuffer = arrowbyteArray.ValueBuffer.Memory;
+                    ReadOnlyMemory<byte> byteNullBitMapBuffer = arrowbyteArray.NullBitmapBuffer.Memory;
+                    dataFrameColumn = new ByteDataFrameColumn(field.Name, byteValueBuffer, byteNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                    break;
+                case ArrowTypeId.UInt16:
+                    PrimitiveArray<ushort> arrowUshortArray = (PrimitiveArray<ushort>)arrowArray;
+                    ReadOnlyMemory<byte> ushortValueBuffer = arrowUshortArray.ValueBuffer.Memory;
+                    ReadOnlyMemory<byte> ushortNullBitMapBuffer = arrowUshortArray.NullBitmapBuffer.Memory;
+                    dataFrameColumn = new UInt16DataFrameColumn(field.Name, ushortValueBuffer, ushortNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                    break;
+                case ArrowTypeId.UInt32:
+                    PrimitiveArray<uint> arrowUintArray = (PrimitiveArray<uint>)arrowArray;
+                    ReadOnlyMemory<byte> uintValueBuffer = arrowUintArray.ValueBuffer.Memory;
+                    ReadOnlyMemory<byte> uintNullBitMapBuffer = arrowUintArray.NullBitmapBuffer.Memory;
+                    dataFrameColumn = new UInt32DataFrameColumn(field.Name, uintValueBuffer, uintNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                    break;
+                case ArrowTypeId.UInt64:
+                    PrimitiveArray<ulong> arrowUlongArray = (PrimitiveArray<ulong>)arrowArray;
+                    ReadOnlyMemory<byte> ulongValueBuffer = arrowUlongArray.ValueBuffer.Memory;
+                    ReadOnlyMemory<byte> ulongNullBitMapBuffer = arrowUlongArray.NullBitmapBuffer.Memory;
+                    dataFrameColumn = new UInt64DataFrameColumn(field.Name, ulongValueBuffer, ulongNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                    break;
+                case ArrowTypeId.Struct:
+                    StructArray structArray = (StructArray)arrowArray;
+                    StructType structType = (StructType)field.DataType;
+                    IEnumerator<Field> fieldsEnumerator = structType.Fields.GetEnumerator();
+                    IEnumerator<IArrowArray> structArrayEnumerator = structArray.Fields.GetEnumerator();
+                    while (fieldsEnumerator.MoveNext() && structArrayEnumerator.MoveNext())
+                    {
+                        AppendDataFrameColumnFromArrowArray(fieldsEnumerator.Current, structArrayEnumerator.Current, ret);
```

Comment:
Should we append the `Struct`'s field name to the beginning of each sub-column's name? That way we won't get column name collisions.

Reply:
Done.